code/modules/bitrunning/server/loot.dm

```diff
@@ -16,11 +16,38 @@
 
 	rewards_base += (length(spawned_threat_refs) * 2)
 
-	for(var/index in 2 to length(avatar_connection_refs))
-		rewards_base += multiplayer_bonus
+	rewards_base += get_multiplayer_bonus()
+
+	rewards_base += get_nohit_bouns()
 
 	return rewards_base
 
+/// Calculates total bonus from completing the domain in multiplayer
+/obj/machinery/quantum_server/proc/get_multiplayer_bonus()
+	var/total = 0
+	var/multiplayer = FALSE
+	for(var/datum/weakref/connection_ref in avatar_connection_refs)
+		var/datum/component/avatar_connection/connection = connection_ref.resolve()
+		if(isnull(connection))
+			continue
+		if(multiplayer)
+			total += multiplayer_bonus
+		multiplayer = TRUE
+	return total
+
+/// Calculates total bonus from completing the domain without taking damage
+/obj/machinery/quantum_server/proc/get_nohit_bouns()
+	if(generated_domain.domain_flags & DOMAIN_NO_NOHIT_BONUS)
+		return 0
+
+	var/total = 0
+	for(var/datum/weakref/connection_ref in avatar_connection_refs)
```

Comment:
can't we throw `as anything` here or does the list contain entries of other types as well?

Reply:
I don't trust it